table.csv

```diff
@@ -530,6 +530,7 @@ skynet-ns,                      namespace,      0xb19910,       draft,      Skyn
 arweave-ns,                     namespace,      0xb29910,       draft,      Arweave Namespace
 subspace-ns,                    namespace,      0xb39910,       draft,      Subspace Network Namespace
 kumandra-ns,                    namespace,      0xb49910,       draft,      Kumandra Network Namespace
+poseidon2-alt_bn_128-a2-cdx1    multihash,      0xcd0001,       draft,      Poseidon2 using BN254 (alt-bn-128) and arity of 2 with Codex parameters
```

Comment:
Well with the current plan we will use both sponge (on the cell level, because it seems it can be faster with the same parameters) and Merkle root (of the cell hashes). So if we look at this a hash of bytestrings (which I guess what this table is about?), then maybe our hash function is really: 

"chunk into 2kb cells, split that into field elements, apply padding, compute the hash with a sponge with given conventions; then build an arity-2 Merkle tree on top of these with this-and-this padding convention"

Quite a mouthful... And of course the 2kb is just a suggestion here. 

And then we probably want another extra padding complication: Namely, first chunk into 64kb blocks, apply the above, then build a Merkle tree on the top of those with given padding convention (the difference between this and the above is padding; but we may want this so that the network-level Merkle root is the same as the root or subroot used for sampling...)

So it's not clear to me how to insert these rather custom solutions into the global table. The simplest is probably just fix 1) a sponge and 2) an arity-2 Merkle root with some well-documented padding conventions; but this does not fully encode the above structures. Or maybe even skip the sponge because it will only appear internally?

Reply:
Actually, I would not overcomplicate things atm (sorry for the initial noise!). 

As long as it satisfies our needs, i.e. pick the correct hashing method based on the CID/multihash, we're good. We can split this up as we see fit before we push upstream, which I don't think is happening any time soon.

Reply:
Ok with me after all. Probably we will change things before pushing upstream anyway.